src/qilib/utils/storage/mongo.py

```diff
@@ -250,7 +250,11 @@ def _is_encoded_int(value: str) -> bool:
         if not value.startswith('_integer[') or not value.endswith(']'):
             return False
 
-        return value[len('_integer['):-1].isdigit()
+        try:
```

Comment:
replace this (slow) try/catch with:
```
string_key = value[len('_integer['):-1]
return string_key[1:].isdigit() if string_key[0] == '-' else string_key.isdigit()
```

Reply:
It's actually not that much slower. And surprisingly, with negative integers it's most of the times even slightly faster than your suggestion. I will use your suggestion though, as I think it's more readable.